### PR TITLE
Fix validation errors not calling callbacks

### PR DESCRIPTION
**Pull request by colincasey**
I noticed this issue while upgrading `tough-cookie` in `jsdom` which was causing their test suite to hang. There is a sneaky little bug around our internal validations where, in several methods that accept callbacks, our validations error out directly and never invoke the given callback with the error. This can lead to code that waits for a callback that will never complete. This also affects functions wrapped in `Sync` which leverage callbacks to capture their return values.

The affected methods were:
- `getCookiesSync`
- `getCookieString`
- `getSetCookieStrings`
- `serializeSync`
- `serialize`
- `setCookieSync`
- `setCookie`
- `getCookiesSync`
- `getCookies`

There were also several validation errors hiding in support methods. These appear to be redundant since the type signatures present already do much of the validation. These have been removed from:
- `checkSameSiteContext`
- `isHostPrefixConditionMet`
- `isSecurePrefixConditionMet`
- `cookieCompare`
- `formatDate`
- `permutePath`

### CODE REVIEW

File: lib/cookie/cookieJar.ts [hunk 1]
```diff
@@ -330,12 +327,22 @@ export class CookieJar {
...
+      })
+    } catch (err) {
+      if (err instanceof Error) {
+        syncErr = err
+      } else {
+        throw err
+      }
+    }
+
     // These seem to be false positives; it can't detect that the value may be changed in the callback
```
Comment (wjhsf) [inline lib/cookie/cookieJar.ts:344]:
```suggestion
```
Comment is for the removed `// eslint-disable`

File: lib/cookie/cookieJar.ts [hunk 2]
```diff
@@ -330,12 +327,22 @@ export class CookieJar {
...
+      fn.call(this, (error: Error | null, result?: T | undefined) => {
+        syncErr = error
+        syncResult = result
+      })
+    } catch (err) {
+      if (err instanceof Error) {
+        syncErr = err
+      } else {
+        throw err
+      }
```
Comment (wjhsf) [inline lib/cookie/cookieJar.ts:341]: Elsewhere we just cast `err as Error`, why be particular here?

**Review by wjhsf — APPROVED** [review]

File: lib/cookie/cookieJar.ts (hunk 1 above)
Comment (colincasey) [reply lib/cookie/cookieJar.ts]: Fixed

File: lib/cookie/cookieJar.ts (hunk 2 above)
Comment (colincasey) [reply lib/cookie/cookieJar.ts]: Changed to cast.

**Review by wjhsf — APPROVED** [review]

**Pull request merged**